### ISSUE TITLE
sc2115 should not warn when "nounset"  is set

### ISSUE THREAD

**Issue by pikrzysztof** [koalaman/shellcheck#2373]
#### For bugs
- Rule Id: 2115 
- My shellcheck version (`shellcheck --version` or "online"):  online and 0.7.2
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash
set -o nounset
x="something"
rm -rf "$x/home"
```

#### Here's what shellcheck currently says:
```
Line 4:
rm -rf "$x/home"
       ^-- SC2115 (warning): Use "${var:?}" to ensure this never expands to /home .
```
#### Here's what I wanted or expected to see:
no warning

In case SC2115 is raised correctly, tell the user that setting `nounset` is also a valid way to fix SC2115.

**Comment by koalaman** [comment]
`nounset` is not a good substitute because it does not protect against set but empty variables. For example, the STEAMROOT bug mentioned on the wiki would still have triggered with `nounset` enabled because the variable was empty due to a failed command expansion.

**Comment by pikrzysztof** [comment]
Hm, I haven't thought about this. Thanks for teaching me something new!
